c7n/resources/ssm.py

```diff
@@ -601,6 +601,25 @@ def register_resource(cls, registry, resource_class):
 resources.subscribe(PostItem.register_resource)
 
 
+class SSMDocumentDescribe(DescribeSource):
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client('ssm')
+
+        def _augment(r):
+            # List tags for the Notebook-Instance & set as attribute
```

Comment:
as noted in the community call, this should really be more of a separate content filter on documents, rather then augment to avoid extra api calls for use cases where the content is not the focus, ie tag policies for example.